src/main/java/com/adobe/cq/testing/client/CQConfigManagerClient.java

```diff
@@ -0,0 +1,235 @@
+    private static final long DEFAULT_TIMEOUT = Duration.ofSeconds(30).toMillis();
+    private static final String GROUPID_TEMPLATE_AUTHORS = "template-authors";
+    private static final String GROUPID_CONTENT_AUTHORS = "content-authors";
+    private static final String GROUPID_EVERYONE = "everyone";
+
+    public CQConfigManagerClient(final CloseableHttpClient http, final SlingClientConfig config) throws ClientException {
+        super(http, config);
+    }
+
+    public CQConfigManagerClient(final URI serverUrl, final String user, final String password) throws ClientException {
+        super(serverUrl, user, password);
+    }
+
+    /** Creates a config on default config path /conf
+     *
+     * @param configTitle title of the config to be created
+     * @param capabilities capabilities of the configs eg Editable Templates
+     * @return created config
+     * @throws ClientException
+     */
+    public CQConfig create(final String configTitle, final CQConfigCapability... capabilities) throws ClientException {
+        return create(DEFAULT_ROOT_PATH, configTitle, configTitle, capabilities);
+    }
+
+    /**
+     * Created a  Config on the given path
+     * @param parentPath path below which config should be created
+     * @param configTitle title of the config to be created
+     * @param capabilities capabilities of the configs eg Editable Templates
+     * @return created config
+     * @throws ClientException
+     */
+    public CQConfig create(final String parentPath, final String configTitle, final CQConfigCapability... capabilities) throws ClientException {
+        return create(parentPath, configTitle, configTitle, capabilities);
+    }
+
+    /**
+     * Created a  Config on the given path and with given name
+     * @param parentPath path below which config should be created
+     * @param configTitle title of the config to be created
+     * @param configName node name for the config
+     * @param capabilities capabilities of the configs eg Editable Templates
+     * @return created config
+     * @throws ClientException
+     */
+    public CQConfig create(final String parentPath, final String configTitle, final String configName, final CQConfigCapability... capabilities) throws ClientException {
+        FormEntityBuilder formEntry = FormEntityBuilder.create()
+                .addParameter(PARAMETER_CHARSET, CHARSET_UTF8)
+                .addParameter(PARAM_TITLE, configTitle)
+                .addParameter(PARAM_NAME, configName)
+                .addParameter(PARAM_PARENT, parentPath);
+        Arrays.stream(capabilities).forEach(c -> {
+            formEntry.addParameter(PARAM_CONFIG_CAPABILITIES, c.getTitle());
+        });
+        doPost(CREATE_ENDPOINT, formEntry.build(), SC_OK);
+        return new CQConfig(this, parentPath + "/" + configTitle);
+    }
+
+    /**
+     * Helper class to update or delete an existing config.
+     */
+    public final class CQConfig {
```

Comment:
why not having this in a separate file? this would imply creating a new package `configmanager` and adding all the files with the classes in there

Reply:
@volteanu  this was a small utility, just to update or delete a exiting CQConfig, if need some functionality with this then we can create a separate class for this IMO